CRM/Contribute/Form/Contribution/Main.php

```diff
@@ -279,9 +279,12 @@ public function setDefaultValues() {
       }
 
       //set custom field defaults
+      $customFieldType = ['Membership'];
       foreach ($this->_fields as $name => $field) {
         if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($name)) {
-          if (!isset($this->_defaults[$name])) {
+          if (!CRM_Core_BAO_CustomGroup::checkCustomField($customFieldID, $customFieldType)
```

Comment:
```suggestion
          if (!CRM_Core_BAO_CustomGroup::checkCustomField($customFieldID, ['Membership'])
```

Reply:
Simpler

Reply:
The second parameter needed to be passed as ref, submitted a [PR](https://github.com/civicrm/civicrm-core/pull/26101) to remove the reference

Reply:
Sorry, missed that, but great to clean that up anyways.